docfiles/tracking.html

```diff
@@ -64,14 +64,23 @@
             }
         });
 
-        var scriptIdRegex = /(?:S?\d{5}-\d{5}-\d{5}-\d{5})|(?:_[0-9a-zA-Z]{12})/g;
-        // Check if the current page contains a share URL
-        function isScriptPage(url) {
-            return !!url.match(scriptIdRegex);
-        }
-        // Scrub the key (if any) from the URL.
+        // Scrub potential PII from the URL.
         function scrubUrl(url) {
-            return url.replace(scriptIdRegex, "xxxxx-xxxxx-xxxxx-xxxxx");
+            if (!url) return url;
+
+            var scriptIdRegex = /(?:S?\d{5}-\d{5}-\d{5}-\d{5})|(?:_[0-9a-zA-Z]{12})/g;
+            var scrubbedUrl = new URL(url.toLocaleLowerCase().replace(scriptIdRegex, "xxxxx-xxxxx-xxxxx-xxxxx"));
+
+            // Just as a precaution, strip any suspicious query parameters.
+            var blockedParamKeywords = ["username", "password", "token", "pwd"];
+            var searchParams = Array.from(scrubbedUrl.searchParams.keys());
+            for (var param of searchParams) {
+                if (blockedParamKeywords.some(k => param.indexOf(k) != -1)) {
+                    scrubbedUrl.searchParams.delete(param);
+                }
+            }
```

Comment:
```suggestion
            for (var blockedKW of blockedParamKeywords) {
                if (scrubbedUrl.searchParams.has(blockedKW)) {
                    scrubbedUrl.searchParams.delete(blockedKW);
                }
            }
```

maybe a bit clearer, since searchParams is already a map

Reply:
This doesn't quite have the same behavior, since it only checks for exact, full-word matches, whereas the intent is to check for params _containing_ any of the keywords. For example, the recent false positive was on a parameter named "force_modify_password". If I'm understanding your suggestion correctly, that would not get filtered, but if we stick to the indexOf check it would. The downside is, of course, we can't just use the convenient `.has()` function, but I think this will be better at weeding out potential issues.

Reply:
Ah, I see, makes sense